hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java

```diff
@@ -229,21 +227,22 @@ public class RocksDBCheckpointDiffer implements AutoCloseable,
             TimeUnit.MILLISECONDS);
 
     if (pruneCompactionDagDaemonRunIntervalInMs > 0) {
-      this.executor = Executors.newSingleThreadScheduledExecutor();
-      this.executor.scheduleWithFixedDelay(
+      this.scheduler = new Scheduler("CompactionDagPruningService",
```

Comment:
BTW had a doubt can't we just have this schedule in a run method or something and still keep this object singleton?

Reply:
Any particular reason why we need to move away from singleton here?

Reply:
Reverted back to singleton and added start/stop methods.